vyos/vyos-1x
# T6246: basic haproxy http-check configuration

Enable http-check configuration for load-balancing revers-proxy.
I'm pretty new to Vyos and this is my first attempt to contribute and hope to get some feedback

## Change Summary
Allow configuration of http-check for haproxy backends

## Types of changes

## Related PR(s)
vyos/vyos-documentation#1388

## Component(s) name
load-balancing

## Proposed changes
Enable configuration options for haproxy backend http-check

## How to test
```
set load-balancing reverse-proxy backend master http-check method 'get'
set load-balancing reverse-proxy backend master http-check uri '/health'
set load-balancing reverse-proxy backend master http-check expect 'status 200'
```

## Smoketest result
```
test_01_lb_reverse_proxy_domain (__main__.TestLoadBalancingReverseProxy.test_01_lb_reverse_proxy_domain) ... ok
test_02_lb_reverse_proxy_cert_not_exists (__main__.TestLoadBalancingReverseProxy.test_02_lb_reverse_proxy_cert_not_exists) ... 
PKI does not contain any certificates!


Certificate "cert" not found in configuration!

ok
test_03_lb_reverse_proxy_ca_not_exists (__main__.TestLoadBalancingReverseProxy.test_03_lb_reverse_proxy_ca_not_exists) ... ok
test_04_lb_reverse_proxy_backend_ssl_no_verify (__main__.TestLoadBalancingReverseProxy.test_04_lb_reverse_proxy_backend_ssl_no_verify) ... 
backend bk-01 cannot have both ssl options no-verify and ca-certificate
set!

ok
test_05_lb_reverse_proxy_backend_http_check (__main__.TestLoadBalancingReverseProxy.test_05_lb_reverse_proxy_backend_http_check) ... ok

----------------------------------------------------------------------
Ran 5 tests in 76.137s

OK
```

## Checklist:
